pkg/utils/utils.go

```diff
@@ -21,3 +25,32 @@ func DecodeSlice(input []string) []string {
 	}
 	return decodedSlice
 }
+
+//EqualOnly compares fields defined in the fields array
+func EqualOnly(sub1 interface{}, sub2 interface{}, fields []string) bool {
+	val1 := reflect.ValueOf(sub1).Elem()
+	val2 := reflect.ValueOf(sub2).Elem()
+	for i := 0; i < val1.NumField(); i++ {
+		typeField := val1.Type().Field(i)
+		if !contains(typeField.Name, fields) {
+			continue
+		}
+		value1 := val1.Field(i)
+		value2 := val2.Field(i)
+		value1 = reflect.NewAt(value1.Type(), unsafe.Pointer(value1.UnsafeAddr())).Elem()
+		value2 = reflect.NewAt(value2.Type(), unsafe.Pointer(value2.UnsafeAddr())).Elem()
```

Comment:
@sakshi-goyal-razorpay moving to test file only since i am using unsafe pointers to excess private fields it is not recomended to put out side test file